### PR TITLE
Fix bug when connecting to multiple servers

### PR DESCRIPTION
**Pull request by Rdornier**
Hello @dominikl, @jburel 

I have an unexpected behavior when I'm connecting to 2 different omero servers.
First, I'm connecting to one server (e.g `server1`) ; the gateway and context are initialized correctly. Then, I get for the `AdminFacility `and `BrowseFacility` instance to retreive some images and check if the current user is admin or not.

Then, I'm connecting to the second server  (e.g `server2`). A new gateway is initiliazed, as well as a new Security context. Everything is fine until I get the `AdminFacility`. At that point, a connection error is raised, which is a bit unexpected as the ``gateway.connect(cred)`` works fine. 

The cause of the connection error seems to be the hostname. When connecting to `server2`, 
- before calling ``gateway.getFacility(AdminFacility.class)``, the hostname in the attached security context is ``server2``
- after calling ``gateway.getFacility(AdminFacility.class)``, the hostname in the attached security context is ``server1``

It takes me quite some time to understand why the swapping is happening but I finally manage to find what causes this issue.
On this [line](https://github.com/ome/omero-gateway-java/blob/master/src/main/java/omero/gateway/facility/Facility.java#L119), the requested facilities are got from the cache (which is a good thing by the way) but they are got only based on their name ; there is no information about the server from which they come from. So, by default, when I request the ``AdminFacility`` on ``server2``, it returns me the ``AdminFacility`` from ``server1`` i.e. the first cached object.

So, I just added, in the key, the hostname of the server and everything worked correctly. I also added a public method to get the hostname from the `Gateway` object.

Rémy

### CODE REVIEW

File: src/main/java/omero/gateway/facility/Facility.java [hunk 1]
```diff
@@ -115,8 +115,14 @@ public static <T extends Facility> T getFacility(final Class<T> type,
                         + type.getSimpleName(), e);
             }
         }
-        
-        return (T) cache.get(type.getSimpleName(), new Callable<Facility>() {
+        String serverHost;
+        try{
+            serverHost = gateway.getServerHost();
+        }catch (DSOutOfServiceException e){
+            throw new ExecutionException("Not log-in ; Can't instantiate "
```
Comment (dominikl) [inline src/main/java/omero/gateway/facility/Facility.java:122]:
```suggestion
            throw new ExecutionException("Not logged in; Can't instantiate "
```

**Comment by dominikl** [comment]
Thank you very much for your contributions @Rdornier . I'd just slightly change the error message, apart from that I think it's good to merge? Any objections @jburel ?

**Comment by Rdornier** [comment]
Thanks for your quick validation !

**Review by dominikl — APPROVED** [review]
👍

**Pull request merged**